lib/live_select/component.ex

```diff
@@ -514,19 +527,38 @@ defmodule LiveSelect.Component do
     })
   end
 
-  defp update_selection(nil, _current_selection, _options, _mode, _value_mapper), do: []
+  defp set_selection(nil, _current_selection, _options, _mode, _value_mapper), do: []
 
-  defp update_selection(value, current_selection, options, :single, value_mapper) do
+  defp set_selection(value, current_selection, options, :single, value_mapper) do
     List.wrap(normalize_selection_value(value, options ++ current_selection, value_mapper))
   end
 
-  defp update_selection(value, current_selection, options, :tags, value_mapper) do
+  defp set_selection(value, current_selection, options, :tags, value_mapper) do
     value = if Enumerable.impl_for(value), do: value, else: [value]
 
     Enum.map(value, &normalize_selection_value(&1, options ++ current_selection, value_mapper))
     |> Enum.reject(&is_nil/1)
   end
 
+  defp update_selection(update_fn, current_selection, options, _mode, value_mapper)
+       when is_function(update_fn, 1) do
+    new_selection = update_fn.(current_selection)
+
+    {existing, new} = Enum.split_with(new_selection, &(&1 in current_selection))
+
+    new =
+      Enum.map(new, &normalize_selection_value(&1, options, value_mapper))
+      |> Enum.reject(&is_nil/1)
+
+    Enum.uniq(existing ++ new)
```

Comment:
This forces uniqueness. Otherwise appending would allow duplicate values and I couldn't think of a reason to allow it.

Reply:
Let's remove this new function, see comment above